src/super_gradients/common/auto_logging/console_logging.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+from io import StringIO
+import atexit
+
+from super_gradients.common.environment.env_helpers import multi_process_safe, is_distributed
+
+
+FILE_BUFFER_SIZE = 10_000
+
+
+class BufferWriter:
+    """File writer buffer that opens a file only when flushing and under the condition that threshold buffersize was reached."""
+
+    def __init__(self, filename: str, buffer_size: int):
+        self.buffer = StringIO()
```

Comment:
please add documentation. is buffer size in MB or in number of chars?

Reply:
Added in 8bfd183beb1ba34012bd58dd0ae3c3a91cf164c1 (number of chars)